wdl-lint/src/rules/command_shellcheck.rs

```diff
@@ -0,0 +1,357 @@
+                    for (_, start, _) in lines_with_offset(text.as_str()) {
+                        // Check to see if we are still on the previous line
+                        // This happens after we encounter a placeholder
+                        if on_same_line {
+                            on_same_line = false;
+                            last_line = last_line.saturating_sub(1);
+                        }
+                        line_map.insert(line_num + last_line, (part.clone(), start));
+                        last_line += 1;
+                    }
+                    line_num += last_line;
+                }
+                CommandPart::Placeholder(_) => {
+                    on_same_line = true;
+                }
+            }
+        }
+
+        match run_shellcheck(&sanitized_command) {
+            Ok(comments) => {
+                for comment in comments {
+                    // skip declarations that shellcheck is unaware of
+                    if comment.code == 2154
+                        && decls.contains(comment.message.split_whitespace().next().unwrap_or(""))
+                    {
+                        continue;
+                    }
+                    let (rel_part, start) = line_map
+                        .get(&comment.line)
+                        .expect("shellcheck line corresponds to command line");
+                    let rel_token = match rel_part {
+                        CommandPart::Text(text) => text.syntax(),
+                        CommandPart::Placeholder(phold) => {
+                            &support::token(phold.syntax(), SyntaxKind::PlaceholderNode)
+                                .expect("should have a placeholder node token")
+                        }
+                    };
+                    let inner_span = {
+                        let outer_span = rel_token.text_range().to_span();
+                        Span::new(
+                            outer_span.start() + leading_whitespace + start + comment.column - 1,
+                            comment.end_column - comment.column,
+                        )
+                    };
+                    state.exceptable_add(
+                        shellcheck_lint(&comment, inner_span),
+                        SyntaxElement::from(rel_token.clone()),
+                        &self.exceptable_nodes(),
+                    )
+                }
+            }
+            Err(e) => {
+                let command_keyword = support::token(section.syntax(), SyntaxKind::CommandKeyword)
+                    .expect("should have a command keyword token");
+                state.exceptable_add(
+                    Diagnostic::error("failed to run shellcheck on command block")
+                        .with_label(e.to_string(), command_keyword.text_range().to_span())
+                        .with_rule(ID)
+                        .with_fix("ensure shellcheck is installed or disable this lint"),
```

Comment:
Does the check above for `command_exists()` not silently return if `shellcheck` isn't present? If so, I'm a bit confused on this one.

Reply:
If we do keep this error message, `shellcheck` needs to be in backticks.

Reply:
It does silently return above if shellcheck isn't installed. The label for this diagnostic will be whatever anyhow error `run_shellcheck` returns. The `fix` message is just poor wording and should be updated.